rest/src/test/scala/org/apache/gearpump/services/AppMasterServiceSpec.scala

```diff
@@ -44,6 +46,16 @@ class AppMasterServiceSpec extends FlatSpec with ScalatestRouteTest with AppMast
       null
   }
 
+  def dag(config: Option[UserConfig]): DAG = {
+    config match {
```

Comment:
We would need an implicit ActorSystem
```  def getValue[T](key: String)(implicit system: ActorSystem): Option[T]  = {
```
So your approach wouldn't work. I agree we shouldn't need to repeat the dag operation within the Spec class.